packages/perseus-editor/src/widgets/interactive-graph-editor/locked-figures/locked-line-settings.tsx

```diff
@@ -73,13 +74,34 @@ const LockedLineSettings = (props: Props) => {
 
     // Check if the line has length 0.
     const isInvalid = kvector.equal(point1.coord, point2.coord);
-
-    function getPrepopulatedAriaLabel() {
+    /**
+     * Generate a prepopulated aria label for the line, with the math
+     * details converted into spoken words.
+     */
+    async function getPrepopulatedAriaLabel() {
         let visiblelabel = "";
+        let point1VisibleLabel = "";
+        let point2VisibleLabel = "";
+
         if (labels && labels.length > 0) {
             visiblelabel += ` ${labels.map((l) => l.text).join(", ")}`;
         }
-        let str = `${capitalizeKind}${visiblelabel} from (${point1.coord[0]}, ${point1.coord[1]}) to (${point2.coord[0]}, ${point2.coord[1]})`;
+
+        if (point1.labels && point1.labels.length > 0) {
+            point1VisibleLabel += ` ${point1.labels
+                .map((l) => l.text)
+                .join(", ")}`;
+        }
+
+        if (point2.labels && point2.labels.length > 0) {
+            point2VisibleLabel += ` ${point2.labels
+                .map((l) => l.text)
+                .join(", ")}`;
+        }
+
+        let str = await generateSpokenMathDetails(
+            `${capitalizeKind}${visiblelabel} from point${point1VisibleLabel} at (${point1.coord[0]}, ${point1.coord[1]}) to point${point2VisibleLabel} at (${point2.coord[0]}, ${point2.coord[1]})`,
+        );
```

Comment:
I think there might be a bug here when the labels include dollar signs.

Because we're calling `generateSpokenMathDetails` on the concatenated string, an unescaped dollar sign in one label might pair with an unescaped dollar sign in another label to create a TeX expression where there shouldn't be one.

For example, if one point had the label `$1` and another point had the label `$2`, the string passed to `generateSpokenMathDetails` would be:

```
Line from point $1 at (0, 0) to point $2 at (1, 1)
```

And that would get interpreted as containing a TeX expression `1 at (0, 0) to point` which is not going to speechify nicely (because every letter will be interpreted as a separate variable).

We could fix this by calling generateSpokenMathDetails for each label individually. For the example above, that would mean we'd get TeX syntax errors for both labels due to the unescaped dollar signs.

That makes me wonder: how are we handling TeX syntax errors?

Reply:
A few points:
1. I don't think you can use unescaped dollar signs as labels, so I'm not sure how to recreate the case you're referring to.
2. I can still call `generateSpokenMathDetails` on each label individually. The reason I didn't do this to begin with is that if it's a math block, it doesn't include the space before, but if it's regular text, it is. So I'd have to add an extra bit of logic to handle this case.
3. TeX syntax errors show up really ugly in the preview with big red text. We don't handle them specifically.

<img width="718" alt="image" src="https://github.com/user-attachments/assets/913e0200-cdd4-49d5-94ff-222507d16ff1">

Reply:
Interesting! Can we file a bug to at some point add an error message into the form for `Visible labels`? We don't need to inflate the scope of your work but I think this is a small change that would make this error more clear to content creators! ❤️